sdk/core/Azure.Core/src/Serialization/AbstractTypeDeserializerAttribute.cs

```diff
@@ -12,23 +12,23 @@ namespace Azure.Core.Serialization
     /// and return an instance of that subtype.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class)]
-    public sealed class AbstractHierarchyDeserializerAttribute : Attribute
+    public sealed class AbstractTypeDeserializerAttribute : Attribute
     {
         /// <summary>
-        /// Instantiates a new instance of the <see cref="AbstractHierarchyDeserializerAttribute"/> class.
+        /// Instantiates a new instance of the <see cref="AbstractTypeDeserializerAttribute"/> class.
         /// </summary>
-        /// <param name="typeToActivate">The <see cref="Type"/> to activate and call deserialize on.</param>
-        public AbstractHierarchyDeserializerAttribute([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)] Type typeToActivate)
+        /// <param name="deserializationProxy">The <see cref="Type"/> to create and call deserialize on.</param>
```

Comment:
I'd suggest mentioning the need for a parameterless constructor here as well.  Given that this is an attribute, I suspect few people will read the docs for the property and will mainly see intellisense for the constructor.